### PR TITLE
Initialize counters from ScioContext 

### PR DESCRIPTION
**Pull request by anish749**
Add a function to initialise multiple counters from ScioContext.

We mostly have this code copied in multiple internal repos and thought of moving it upstream.

### CODE REVIEW

File: scio-test/src/test/scala/com/spotify/scio/ScioContextTest.scala [hunk 1]
```diff
@@ -175,4 +175,42 @@ class ScioContextTest extends PipelineSpec {
     the[IllegalArgumentException] thrownBy { ScioContext.apply(invalidOpts) } should have message
       s"blockFor param foo cannot be cast to type scala.concurrent.duration.Duration"
   }
+
```
Comment (anish749) [inline scio-test/src/test/scala/com/spotify/scio/ScioContextTest.scala]: are these test cases needed? I saw that the part was already covered in codecov, but couldn't find where the test case was !!

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 2]
```diff
@@ -757,13 +757,19 @@ class ScioContext private[scio] (val options: PipelineOptions, private var artif
...
+   * name.
+   * */
   def initCounter(namespace: String, name: String): Counter = {
     val counter = ScioMetrics.counter(namespace, name)
     _counters.append(counter)
     counter
   }
 
+  /** Initialize a given [[org.apache.beam.sdk.metrics.Counter Counter]] metric. */
+  def initCounter(counter: Counter*): Unit =
```
Comment (regadas) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: @anish749 I think i would do this

```suggestion
  def initCounter(counter: Counter*): List[Counter] =
```

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 3]
```diff
@@ -757,13 +757,19 @@ class ScioContext private[scio] (val options: PipelineOptions, private var artif
...
   }
 
-  /** Initialize a new [[org.apache.beam.sdk.metrics.Counter Counter]] metric. */
+  /**
+   * Initialize a new [[org.apache.beam.sdk.metrics.Counter Counter]] metric from namespace and
+   * name.
+   * */
   def initCounter(namespace: String, name: String): Counter = {
     val counter = ScioMetrics.counter(namespace, name)
     _counters.append(counter)
```
Comment (regadas) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: you can replace this with the new call

File: scio-test/src/test/scala/com/spotify/scio/ScioContextTest.scala (hunk 1 above)
Comment (regadas) [reply scio-test/src/test/scala/com/spotify/scio/ScioContextTest.scala]: I don't think we need them. If you replace the internal call with the new one.

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala (hunk 3 above)
Comment (anish749) [reply scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: Yes, thats a good idea. I'll do that

**Comment by regadas** [comment]
Thx @anish749!

**Pull request merged**